From d20fb18743cf3cfeb879329e8c3551b0d0c258d7 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Tue, 12 May 2015 12:11:19 +0800
Subject: [PATCH] export.h's: Adapt for Symbol Export on Visual Studio

Add code paths for exporting and importing symbols and variables on Visual
Studio, so we do not need to rely on .def files and public variables can
be correctly exported and referred to.
---
 include/export.h            | 19 +++++++++++++++----
 libcharset/include/export.h | 19 +++++++++++++++----
 2 files changed, 30 insertions(+), 8 deletions(-)

diff --git a/include/export.h b/include/export.h
index 62fb77b..7e4fec0 100644
--- a/include/export.h
+++ b/include/export.h
@@ -1,6 +1,17 @@
-
-#if @HAVE_VISIBILITY@ && BUILDING_LIBICONV
-#define LIBICONV_DLL_EXPORTED __attribute__((__visibility__("default")))
+#ifndef _MSC_VER
+# if @HAVE_VISIBILITY@ && BUILDING_LIBICONV
+#  define LIBICONV_DLL_EXPORTED __attribute__((__visibility__("default")))
+# else
+#  define LIBICONV_DLL_EXPORTED
+# endif
 #else
-#define LIBICONV_DLL_EXPORTED
+# ifndef ICONV_STATIC
+#  if BUILDING_LIBICONV
+#   define LIBICONV_DLL_EXPORTED __declspec(dllexport)
+#  else
+#   define LIBICONV_DLL_EXPORTED __declspec(dllimport)
+#  endif
+# else
+#  define LIBICONV_DLL_EXPORTED
+# endif
 #endif
diff --git a/libcharset/include/export.h b/libcharset/include/export.h
index 84e74aa..7208dbc 100644
--- a/libcharset/include/export.h
+++ b/libcharset/include/export.h
@@ -1,6 +1,17 @@
-
-#if @HAVE_VISIBILITY@ && BUILDING_LIBCHARSET
-#define LIBCHARSET_DLL_EXPORTED __attribute__((__visibility__("default")))
+#ifndef _MSC_VER
+# if @HAVE_VISIBILITY@ && BUILDING_LIBCHARSET
+#  define LIBCHARSET_DLL_EXPORTED __attribute__((__visibility__("default")))
+# else
+#  define LIBCHARSET_DLL_EXPORTED
+# endif
 #else
-#define LIBCHARSET_DLL_EXPORTED
+# ifndef ICONV_STATIC
+#  if BUILDING_LIBCHARSET
+#    define LIBCHARSET_DLL_EXPORTED __declspec(dllexport)
+#  else
+#    define LIBCHARSET_DLL_EXPORTED __declspec(dllimport)
+#  endif
+# else
+#  define LIBCHARSET_DLL_EXPORTED
+# endif
 #endif
-- 
2.3.5.windows.8

